test/parser.test.coffee

```diff
@@ -574,3 +558,14 @@ module.exports =
     console.log 'Result object: ' + util.inspect r, false, 10
     equ r.hasOwnProperty('SAMP'), true
     equ r.SAMP.hasOwnProperty('TAGN'), true)
+
+  'test return a promise': (test)->
+    xml = '<xml></xml>'
+    ret = xml2js.parseString(xml)
+    assert.ok ret instanceof Promise
+    ret
+      .then (r) ->
+        assert.ok r.hasOwnProperty 'xml'
+        test.finish()
+      .catch (err) ->
+        assert.ok false
```

Comment:
I would like to see a test that shows that invalid XML will return a Promise that is rejected.